services/app-api/src/postgres/contractAndRates/findContractWithHistory.ts

```diff
@@ -10,9 +11,9 @@ import { updateInfoIncludeUpdater } from '../prismaHelpers'
 async function findContractWithHistory(
     client: PrismaTransactionType,
     contractID: string
-): Promise<ContractType | Error> {
+): Promise<ContractType | StoreError | Error> {
```

Comment:
This strategy is documented in an ADR: https://github.com/Enterprise-CMCS/managed-care-review/blob/main/docs/architectural-decision-records/012-custom-error-types.md

Note that in that ADR it shows calling Object.setPrototypeOf(this, FooError.prototype); instead of setting name, I'd do some googling to see what is required to do this right.

Reply:
Thanks! I'll make these changes.